viewing_party/party.py

```diff
@@ -1,23 +1,161 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
+    if not title or not genre or not rating :
+        return None 
+    else:
+        new_movie["title"] = title
+        new_movie["genre"] = genre
+        new_movie["rating"] = rating
+        return new_movie
+    
+    
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data 
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data 
+
+def watch_movie(user_data,title):
+
+    for movie in user_data["watchlist"] :
+
+        if title in movie["title"] :
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+    return user_data    
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    if not user_data["watched"]:
+        return 0.0
+    
+    total_rating = 0.0
+    for movie in user_data["watched"]:
+        total_rating += movie["rating"]
+    average_rating = total_rating/len(user_data["watched"])   
+    return average_rating
+        
+def get_most_watched_genre(user_data):
+    if not user_data["watched"]:
+        return None
+    genre_dict = {}

```

Comment:
I suggest adding a blank line here to separate our validation code from the next chunk of code that handles creating the `genre_dict`. This kind of visual separation makes it much easier to see the sections of a function with different responsibilities at a glance.
```suggestion
        return None
        
    genre_dict = {}
```